packages/eas-cli/src/commands/worker/alias.ts

```diff
@@ -0,0 +1,119 @@
+import { Flags } from '@oclif/core';
+import chalk from 'chalk';
+
+import EasCommand from '../../commandUtils/EasCommand';
+import { ExpoGraphqlClient } from '../../commandUtils/context/contextUtils/createGraphqlClient';
+import Log from '../../log';
+import { ora } from '../../ora';
+import { promptAsync } from '../../prompts';
+import {
+  assignWorkerDeploymentAliasAsync,
+  selectWorkerDeploymentOnAppAsync,
+} from '../../worker/deployment';
+
+export default class WorkerAlias extends EasCommand {
+  static override description = 'Assign deployment aliases';
+  static override aliases = ['deploy:alias'];
+
+  // TODO(@kitten): Keep command hidden until worker deployments are live
+  static override hidden = true;
+  static override state = 'beta';
+
+  static override flags = {
+    id: Flags.string({
+      description: 'Worker deployment identifier',
+      required: false,
+    }),
+    alias: Flags.string({
+      description: 'Worker deployment alias name to assign',
+      required: false,
+    }),
+  };
+
+  static override contextDefinition = {
+    ...this.ContextOptions.DynamicProjectConfig,
+    ...this.ContextOptions.ProjectDir,
+    ...this.ContextOptions.LoggedIn,
+  };
+
+  override async runAsync(): Promise<void> {
+    Log.warn('EAS Worker Deployments are in beta and subject to breaking changes.');
+
+    const { flags } = await this.parse(WorkerAlias);
+    const {
+      getDynamicPrivateProjectConfigAsync,
+      loggedIn: { graphqlClient },
+    } = await this.getContextAsync(WorkerAlias, {
+      nonInteractive: true,
+    });
+
+    const { projectId } = await getDynamicPrivateProjectConfigAsync();
+    const aliasName = await resolveDeploymentAliasAsync({ flagAlias: flags.alias });
+    const deploymentId = await resolveDeploymentIdAsync({
+      graphqlClient,
+      aliasName,
+      appId: projectId,
+      flagId: flags.id,
+    });
+
+    const progress = ora('Assigning alias to worker deployment').start();
+    const workerAlias = await assignWorkerDeploymentAliasAsync({
+      graphqlClient,
+      appId: projectId,
+      deploymentId,
+      aliasName,
+    });
+
+    progress.succeed(
+      chalk`Alias {bold ${workerAlias.aliasName}} assigned to deployment {bold ${deploymentId}}`
+    );
+
+    const baseDomain = process.env.EXPO_STAGING ? 'staging.expo' : 'expo';
+    const aliasUrl = `https://${baseDomain}.dev/projects/${projectId}/serverless/deployments`;
```

Comment:
I believe we use this domain creation logic in multiple places. Would it make sense to move it to a function?

Reply:
Yeah, that would be a nice cleanup. In this case I just copied it to get this working, its mostly used in 2 places now.